src/main/java/com/snowflake/kafka/connector/internal/SnowflakeTelemetryPipeStatus.java

```diff
@@ -0,0 +1,185 @@
+package com.snowflake.kafka.connector.internal;
+
+import net.snowflake.client.jdbc.internal.fasterxml.jackson.databind.node.ObjectNode;
+
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+public class SnowflakeTelemetryPipeStatus extends SnowflakeTelemetryBasicInfo {
+  // Offset info
+  AtomicLong processedOffset;           // processed offset + 1 (offset that is most recent in buffer)
+  AtomicLong flushedOffset;             // flushed offset + 1 (files on stage)
```

Comment:
Why are we storing this with a +1 ? It's confusing. What exactly is committedOffset ?

Reply:
So if we flushed a file that contains offset from 0-99, then flushed offset will be 100. I think you are right, it's better to either change the name of the variable to "next"someOffset or remove the +1 from all the variables to avoid confusion. 
I think I'll go with removing +1 from all offsets, WDYT?